Mathlib/Algebra/Group/Submonoid/BigOperators.lean

```diff
@@ -138,4 +141,43 @@ theorem noncommProd_mem (S : Submonoid M) {ι : Type*} (t : Finset ι) (f : ι 
   rintro ⟨x, ⟨hx, rfl⟩⟩
   exact h x hx
 
+end Monoid
+
+section CommMonoid
+variable [CommMonoid M] {x : M}
+
+@[to_additive]
+lemma mem_closure_iff_exists_finset_subset {s : Set M} :
+    x ∈ closure s ↔
+      ∃ (n : M → ℕ) (t : Finset M), ↑t ⊆ s ∧ n.support ⊆ t ∧ x = ∏ a ∈ t, a ^ n a where
+  mp hx := by
+    classical
+    induction' hx using closure_induction with x hx x y _ _ hx hy
```

Comment:
Optional: how does this look with `induction` and `|`? In my experience, this kind of closure lemma makes using that structure really nice, eg it puts the names closer to where they're relevant

Reply:
I think this is a good example of where we should indeed use that spelling, even if it's a few lines longer.